chart/templates/servicemonitor.yaml

```
@@ -1,16 +1,24 @@
-{{- if .Values.serviceMonitor -}}
+{{- if .Values.serviceMonitor.enabled }}
```

As already stated, this is technically breaking but also makes the values structure a bit more consistent (all other "addon" features also use the `{{ whatever }}.enabled` format).

To keep at least some compatibility, maybe test for this feature like this (untested)?

```suggestion
{{- if or (.Values.serviceMonitor.enabled (eq .Values.serviceMonitor true)) }}
```

This would at least keep the ServiceMonitor active for users who just have a `serviceMonitor=true` value. What do you think?

---

I'm not sure how doable this is.

Your suggestion gives either error (depending on bool vs. map):
> Error: template: kube-httpcache/templates/servicemonitor.yaml:1:18: executing "kube-httpcache/templates/servicemonitor.yaml" at <.Values.serviceMonitor.enabled>: can't evaluate field enabled in type interface {}

> Error: template: kube-httpcache/templates/servicemonitor.yaml:1:18: executing "kube-httpcache/templates/servicemonitor.yaml" at <.Values.serviceMonitor.enabled>: enabled is not a method but has arguments

I tried adding some type checking logic, but the best I manged was this - but it still fails when it's a `bool`.

```
{{- if or (and (kindIs "bool" .Values.serviceMonitor) .Values.serviceMonitor) (and (kindIs "map" .Values.serviceMonitor) .Values.serviceMonitor.enabled)}}
```

`serviceMonitor: true` gave:
> Error: template: kube-httpcache/templates/servicemonitor.yaml:1:128: executing "kube-httpcache/templates/servicemonitor.yaml" at <.Values.serviceMonitor.enabled>: can't evaluate field enabled in type interface {}

===

The other point is then each usage of the "additionalLabels" / "interval" needs the same type check.

The only other route I can see is adding a new name for the map (`serviceMonitor2` :D ) and then using either bool value in the OR check?

Maybe you can see a better way / improvement to my templating above?

---

Sorry for the late response. Adding a new `serviceMonitor2` key does not feel like a satisfying solution to me. Maybe we can afford a tiny breaking change (especially given that we're technically still at a 0.something version).

---

No worries, we appreciate the time you put in